couchbase/io/http_session_manager.hxx

```diff
@@ -64,6 +64,7 @@ class http_session_manager : public std::enable_shared_from_this<http_session_ma
     void set_configuration(const topology::configuration& config, const cluster_options& options)
     {
         options_ = options;
+        std::scoped_lock index_lock(next_index_mutex_);
```

Comment:
Makes sense.   Only worry I have is nested mutexes - if the config_mutex_ is ever locked, _then_ the next_index_mutex_ is locked, we could get a deadlock.  As long as that never happens (like, maybe this is the only place we take both?), then cool.

Reply:
I've updated pr, to use `std::scoped_lock` on both mutexes